### ISSUE TITLE
Does not load the source image when open viewing.

### ISSUE THREAD

**Issue by csfedorov** [bpatrik/pigallery2#951]
**Describe the bug**

First off, thank you so much for your product! Very helpful and helps me out a lot!

**Load full resolution image on zoom** - disabled.

When opening a preview of any image, the source is not loaded, but only the generated preview is displayed.
(but if enable in settings **Load full resolution image on zoom**, then when you zoom in, the original is loaded.)

Expected:
A preview of the original image opens.

Actual:
A preview of the generated image preview opens.

I tried waiting more time after generating previews and reloading the docker container, it didn't help.
I also have a version built from the sources of the master branch from December 2023, there is no this problem.
If this is a feature in the new version, how do you make the source image load when opened in the viewer?

![image](https://github.com/user-attachments/assets/65c15163-0f68-4811-a0eb-cfde670e3143)

**Environment:**
 - Debian 11
 - Browser: chrome 130

**Used app version**:
[edge-debian-bookworm](https://hub.docker.com/layers/bpatrik/pigallery2/edge-debian-bookworm/images/sha256-df0c295fb28986b1d4399d71ba78911b40df10fc9b0bbf1cbeeef05eef1ac684?context=explore) (Last pushed 12 days ago)

**Comment by mrtnbr** [comment]
I see the same behaviour. Sometimes, the original does seem to load automatically. I have unfortunately not had enough time to dig deeper. I am also on the latest edge release of pigallery2, Docker container on Debian. I use Firefox on Debian as a browser. However when using Chromium as a browser, I have the 'feeling' the originals load more often.

**Comment by bencahill** [comment]
I can confirm this bug on edge-debian-bookworm.

I think this bug was introduced with #806 

It looks like the assumption is made that there is a preview size big enough for the lightbox.

But I already have my photos small enough that I want the original image loaded for the lightbox view. I only set it to make one thumbnail size of 240px. But then it keeps using the 240px size in the lightbox which looks bad, until the image is zoomed, and then it loads the original image.

Can it be changed to loading the original image in the lightbox if there are no preview/thumbnail sizes large enough to fill the screen?

Thank you!

Two workarounds for this bug until it is fixed:

1. Downgrade to v2.0.0 which was released in Oct 2023 before the breaking change in Jan 2024.
2. Add a preview size big enough to be sufficient to the list of generated sizes.

Once again, I can't stress enough how amazing this software is! Thank you @bpatrik for all your time developing it. 😊

**Comment by bpatrik** [comment]
Made a commit to fix this. Can you please test it?

**Comment by Tianpapa** [comment]
> Made a commit to fix this. Can you please test it?

still a problem here, while my phone device resolution is 2600x1200, I set "thumbnail and photo preview sizes" to "320;1200;2160", however when I open a test_img which size is 4000x3000, fullsrceen. I checked the tmp folder, only test_img.jpg_320q80cs.webp is generate.

**Comment by bpatrik** [comment]
try disabling this: 

<img width="1513" height="517" alt="Image" src="https://github.com/user-attachments/assets/776dc0f6-8273-431f-a8c0-32a86f55251c" />

**Comment by kennethsible** [comment]
@bpatrik I don't think this setting is working as intended. I generated 320px and 540px thumbnails, but loading a 4K image on my iPad (2388-by-1668 pixels) uses the 540px previews instead of the full-resolution image. Looking at the code:

```javascript
getMediaSize(renderWidth: number, renderHeight: number): number {
    const longerEdge = Math.max(renderWidth, renderHeight);
    return Utils.findClosestinSorted(longerEdge, MediaIcon.sortedThumbnailSizes);
}

private isPreviewAdequateForLightbox(): boolean {
    if (!Config.Gallery.Lightbox.loadFullImageIfPreviewTooSmall) {
      return true;
    }

    const selectedSize = this.gridMedia.getMediaSize(window.innerWidth, window.innerHeight);
    const minDisplaySize = Math.min(window.innerWidth, window.innerHeight);

    // If the selected preview size is less than 50% of the minimum display dimension,
    // consider it inadequate for lightbox display (e.g., 240px on a 1080p screen)
    return selectedSize >= minDisplaySize * 0.5;
}
```

It looks like there's a mismatch between physical pixels and CSS pixels: `selectedSize` uses physical pixels but `minDisplaySize` uses CSS pixels. I think multiplying `minDisplaySize` by `window.devicePixelRatio` would fix this.
